### ISSUE TITLE
lint: checker that detects calls like copy(dst, dst) 

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#399]
Example:
```go
func withWarning() {
	var dst []int
	var rw readWriter

	/// suspicious call to copy in `copy(dst, dst)`
	copy(dst, dst)
	/// suspicious call to copy in `copy(dst[1:], dst[1:])`
	copy(dst[1:], dst[1:])

	/// suspicious call to append in `append(dst, dst...)`
	dst = append(dst, dst...)

	/// suspicious call to io.Copy in `io.Copy(rw, rw)`
	io.Copy(rw, rw)
}
```

Do we want such kind of checker?
Should it only handle `copy` or more functions, like in example above (How to avoid false positives when then?).

**Comment by Quasilyte** [comment]
It's aimed to catch `copy(dst, dst)` error that are intended to be `copy(dst, src)`.
In other words, it's copy/paste error checker.

**Comment by Quasilyte** [comment]
Partial implementation:
```go
package lint

import (
	"go/ast"

	"github.com/go-toolsmith/astequal"
)

func init() {
	addChecker(&dupArgChecker{}, attrExperimental)
}

type dupArgChecker struct {
	checkerBase
}

func (c *dupArgChecker) VisitExpr(expr ast.Expr) {
	call, ok := expr.(*ast.CallExpr)
	if !ok {
		return
	}

	args := call.Args
	switch name := qualifiedName(call.Fun); name {
	case "copy", "append", "io.Copy", "io.CopyN", "io.CopyBuffer":
		// Usually expected to be (dst, src) pair.
		if astequal.Expr(args[0], args[1]) {
			c.warn(call, name)
		}
	}
}

func (c *dupArgChecker) warn(cause ast.Node, msg string) {
	c.ctx.Warn(cause, "suspicious call to %s in `%s`", msg, cause)
}

```
(does not include doc comment.)
